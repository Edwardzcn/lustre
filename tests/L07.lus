type T = enum { A, B };

function f(x : int) returns (y : T)
let
  y = if (x > 5) then A else B;
tel
