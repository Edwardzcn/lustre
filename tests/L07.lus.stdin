7
7
5
5
10
10
6
6
11
