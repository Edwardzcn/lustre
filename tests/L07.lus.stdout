node (x : int)
  returns (y)
  assumes ()
  shows ()
let
  y : int
    = ITE(Gt(x, 5), 0, 1)
tel
--- Step 1 ---
x : int  = 7
y = 0
--- Step 2 ---
x : int  = 7
y = 0
--- Step 3 ---
x : int  = 5
y = 1
--- Step 4 ---
x : int  = 5
y = 1
--- Step 5 ---
x : int  = 10
y = 0
--- Step 6 ---
x : int  = 10
y = 0
--- Step 7 ---
x : int  = 6
y = 0
--- Step 8 ---
x : int  = 6
y = 0
--- Step 9 ---
x : int  = 11
y = 0
--- Step 10 ---
x : int  = (EOF)
